crates/proof-of-sql/src/base/math/decimal.rs

```diff
@@ -187,17 +187,20 @@ pub(crate) fn try_into_to_scalar<S: Scalar>(
 /// Scale scalar by the given scale factor. Negative scaling is not allowed.
 /// Note that we do not check for overflow.
 pub(crate) fn scale_scalar<S: Scalar>(s: S, scale: i8) -> DecimalResult<S> {
```

Comment:
See earlier comment. I almost think it would be better to drop this method in favor of a `Scalar::exp10(power: i8) -> Scalar` method. 10 and 1/10 could easily be associated constants on the `Scalar` trait to handle negative powers nicely as well.
Then, we could replace
`scale_scalar(value, scale_factor)` with 
`value * Scalar::exp10(scale_factor)` which is much more readable in my opinion and doesn't need to return a `Result`.